### PR TITLE
fix: [CDS-77293]: relaxed uivalidation for variable name

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

develop pr for the hotfix: 
https://github.com/harness/harness-core-ui/pull/17457/files

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/AddEditCustomVariable.tsx [hunk 1]
```diff
@@ -101,7 +101,7 @@ export default function AddEditCustomVariable(props: AddEditCustomVariableProps)
               getString('common.validation.fieldCannotbeLongerThanN', { name: getString('name'), n: MAX_LENGTH })
             )
             .matches(
-              /^[a-zA-Z_][0-9a-zA-Z_$.]*$/,
+              /^[a-zA-Z_][0-9a-zA-Z-_$.]*$/,
```
Comment (pankajsingh-harness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/AddEditCustomVariable.tsx:104]: Can we use `NameSchema` from validations here?

**Review by cjlee01 — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/AddEditCustomVariable.tsx (hunk 1 above)
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/AddEditCustomVariable.tsx:104]: Please update the error message in the validation to also include `-` as it is allowed now in the regex

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/AddEditCustomVariable.tsx (hunk 1 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/AddEditCustomVariable.tsx:104]: image

image

Can we confirm once with BE, that the regex which BE has to validate variable names is `"^[a-zA-Z_][0-9a-zA-Z_\.$-]{0,127}$"` where numbers are allowed at the end, which our current regex will block

Comment (sahithikolichala) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/CustomVariables/AddEditCustomVariable.tsx:104]: yes confirmed with BE 
and my current regex does

https://github.com/harness/harness-core-ui/assets/57372986/da7acb02-c3eb-4718-9743-593705c4c09b

 not allow the following

**Comment by sahithikolichala** [comment]
modified the error message
Screenshot 2023-08-23 at 1 36 14 PM

**Review by cjlee01 — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**